cmd/entrypoint/secrets.go

```diff
@@ -281,6 +282,21 @@ func ReconcileSecrets(ctx context.Context, sh *SnapshotHolder) error {
 		// ...and for Edge Stack, we _always_ have an implicit reference to the
 		// license secret.
 		secretRef(GetLicenseSecretNamespace(), GetLicenseSecretName(), false, action)
+		// We also want to grab any secrets referenced by edge-stack filters for use in edge-stack
+		// the Filters are unstructured because emissary does not have their type definition
+		for _, f := range sh.k8sSnapshot.Filters {
+			secretName, secretNamespace := findFilterSecret(f)
+			if secretName != "" {
+				if secretNamespace != "" {
```

Comment:
I would move this statement into the `findFilterSecret` func since you are already passing in the Filter and you are already checking whether the namespace exist there. This way you can contain the parsing logic to a single function that can be easily unit tested.

```golang
if secreteName != nil && secretNames != nil { 
    secretRef(secretNamespace, secretName, false, action)
}
```